### PR TITLE
Update README.md in examples folder

### PR DESCRIPTION
**Pull request by sjdilkes**
This README.md file is missing links to new example notebooks and has outdated links to previous example notebooks.

### CODE REVIEW

File: examples/README.md [hunk 1]
```diff
@@ -4,15 +4,20 @@ Here you can find notebooks showing features and example usage of pytket.
 
 *Note: When viewing our examples, github has a bug where it occasionally can't render Jupyter notebooks properly. To bypass this, you can either clone the repo and open the examples in Jupyter yourself (this is recommended! You can play around with them), or you can enter the corresponding URL into https://nbviewer.jupyter.org and view examples there.*
 
-[cirq_routing_example](https://github.com/CQCL/pytket/blob/master/examples/cirq_routing_example.ipynb) - Cirq interface and routing.
+[backends_example](https://github.com/CQCL/pytket/blob/master/examples/backends_example.ipynb) - Running circuits on our allowed backends.
```
Comment (cqc-alec) [inline examples/README.md:7]: How about "some of our backends" rather than "our allowed backends"?

File: examples/README.md [hunk 2]
```diff
@@ -4,15 +4,20 @@ Here you can find notebooks showing features and example usage of pytket.
...
 
+[circuit_generation_example](https://github.com/CQCL/pytket/blob/master/examples/circuit_generation_example.ipynb) - Advance methods for circuit generation in pytket.
```
Comment (cqc-alec) [inline examples/README.md:9]: "Advance" --> "Advanced"

File: examples/README.md [hunk 3]
```diff
@@ -4,15 +4,20 @@ Here you can find notebooks showing features and example usage of pytket.
...
 
-[pyquil_example](https://github.com/CQCL/pytket/blob/master/examples/cirq_routing_example.ipynb) - PyQuil interface and leveraging features across platforms.
+[circuit_analysis_example](https://github.com/CQCL/pytket/blob/master/examples/circuit_analysis_example.ipynb) - Methods for visualizing and anaylsing circuits in pytket.
```
Comment (cqc-alec) [inline examples/README.md:11]: "anaylsing" --> "analysing".

Comment (cqc-alec) [inline examples/README.md:11]: Elsewhere in our docs we quite consistently use trad. British "ise" spellings. Although I personally prefer "ize" spellings, we should be consistent -- so, "visualising" here.

File: examples/README.md (hunk 1 above)
Comment (sjdilkes) [reply examples/README.md]: Makes sense. Have fixed all spelling mistakes.

**Pull request merged**